src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -264,10 +270,70 @@ private int LineBreaksAfter(SyntaxToken currentToken, SyntaxToken nextToken)
             return 0;
         }
 
+        private static bool All<TNode>(SyntaxList<TNode> items, Func<TNode, bool> predicate)
+            where TNode : SyntaxNode
+        {
+            bool result = true;
+            for (int i = 0; i < items.Count; i++)
+            {
+                result &= predicate(items[i]);
+            }
+
+            return result;
+        }
+
+        private static bool IsAutoAccessorList(SyntaxToken token)
+            => token.Parent is AccessorListSyntax accessorList &&
+                All(accessorList.Accessors, a => a.Body == null);
+
+        private static bool IsAutoAccessorList(SyntaxNode? node)
```

Comment:
>Auto [](start = 30, length = 4)

I think the use of the term auto in the name can lead to some confusion. Strictly speaking, we cannot determine if the property is going to be an auto-property just by looking at the accessor list, in fact, we cannot even determine if this is a property we are dealing with. Consider, using a name that more accurately reflects the nature of the check, which is to check if accessors have a body.

Reply:
BTW, are we testing effect of this change on events?

---
In reply to: [530475061](https://github.com/dotnet/roslyn/pull/49495#discussion_r530475061) [](ancestors = 530475061)

Reply:
I don't get why we should not be able to detec whether they are auto implemented properties or not.
As for the initial post, this was my goal:
- collapse the auto implemented properties
- fix the double (wrong) extra line terminators on the initializers
The rest of the formatting should stay the same (and I added a lot of tests on incorrect syntax as well just to cover those cases).
With regards to events, I will comment them on the test question you commented.

Reply:
> I don't get why we should not be able to detect whether they are auto implemented properties or not.

Auto-implemented properties is a very specific term, that is define in language specification. The implementation simply doesn't follow that definition. It appears to accept events and indexers, for example. It appears to accept expression bodied accessors. 

> As for the initial post, this was my goal:

Exactly, you stated this goal, but you change appears to do something else. More specifically, it affects more scenarios. Either you should adjust implementation, or adjust the intent (perhaps expand it) and use terminology according to that.

---
In reply to: [531005203](https://github.com/dotnet/roslyn/pull/49495#discussion_r531005203) [](ancestors = 531005203)

Reply:
Method renamed. Let's see the new tests and then tell me if there are extra changes here.